src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,450 @@
+     */
+    function quit() external onlyMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external onlyMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out `amt` of `token` sitting in this contract to `usr` address.
+     * @dev Can only be called by the admin.
+     * @param token Token address.
+     * @param usr Destination address.
+     * @param amt Token amount.
+     */
+    function yank(
+        address token,
+        address usr,
+        uint256 amt
+    ) external auth {
+        GemAbstract(token).transfer(usr, amt);
+        emit Yank(token, usr, amt);
+    }
+
+    /**
+     * @notice Return Gem address of the selected PSM
+     * @return gem Gem address.
+     */
+    function gem() public view returns (address) {
+        return psm != address(0) ? GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem() : address(0);
+    }
+
+    /**
+     * @notice Calculates the amount of GEM received for swapping `wad` of DAI.
+     * @param wad DAI amount.
+     * @return amt Expected GEM amount.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256 amt) {
+        require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");
+
+        // We ensure the subtraction below will not overflow in `clap`
+        uint256 to18ConversionFactor = 10**(18 - uint256(GemAbstract(gem()).decimals()));
+
+        return (wad * WAD) / ((WAD + PsmAbstract(psm).tout()) * to18ConversionFactor);
+    }
+
+    /**
+     * @notice Calculates the required amount of DAI to get `amt` amount of GEM.
+     * @param amt GEM amount.
+     * @return wad Required DAI amount.
+     */
+    function requiredDaiWad(uint256 amt) external view returns (uint256 wad) {
+        require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");
+
+        // We ensure the subtraction below will not overflow in `clap`
+        uint256 amt18 = amt * 10**(18 - uint256(GemAbstract(gem()).decimals()));
+        uint256 fee = (amt18 * PsmAbstract(psm).tout()) / WAD;
+
+        return amt18 + fee;
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @param wad DAI amount.
+     */
+    function _doPush(uint256 wad) internal {
+        require(to != address(0), "RwaMultiSwapOutputConduit/to-not-picked");
+
+        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
+        uint256 gemAmt = expectedGemAmt(wad);
+        require(gemAmt > 0, "RwaMultiSwapOutputConduit/insufficient-swap-gem-amount");
+
+        address recipient = to;
+        address _psm = psm;
+        address _gem = gem();
```

Comment:
Maybe it is better to just make the getter `external`  just call in the event emission line the `GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem()` directly. The same for `expectedGemAmt ` and `requiredDaiWad` which already are checking for `psm` not being `address(0)`.

Reply:
Looks good, this can be solved.